loader.go

```diff
@@ -29,48 +29,68 @@ func news3Loader(c s3Config) (s3Loader, error) {
 	return s3Loader{*s3Client, c}, nil
 }
 
-func (s3Loader *s3Loader) LoadResource(path string) (io.ReadCloser, error) {
+func (s3Loader *s3Loader) FindLatestResourcesFolder() (string, error) {
 	s3Client := &s3Loader.client
 	if s3Client == nil {
-		return nil, errors.New("S3 bucket not initialised. Please call news3Loader(c s3Config) function first")
+		return "", errors.New("S3 bucket not initialised. Please call news3Loader(c s3Config) function first")
 	}
 
 	doneCh := make(chan struct{})
 	defer close(doneCh)
 
-	r := regexp.MustCompile("[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])")
-	var latestObj = &struct {
-		object minio.ObjectInfo
-		date   time.Time
-	}{}
+	var latestDate time.Time
+
+	for object := range s3Client.ListObjects(s3Loader.config.bucket, "", true, doneCh) {
 
-	for object := range s3Client.ListObjects(s3Loader.config.bucket, financialInstrumentsFolderName, true, doneCh) {
 		if object.Err != nil {
-			return nil, object.Err
+			return "", object.Err
 		}
 
-		if !strings.Contains(object.Key, path) {
+		if strings.Contains(object.Key, ".md5") {
```

Comment:
The reason behind this check is to avoid too many logs such as "Ignoring file [%s]. Cannot parse name." but I can remove it if this is not a problem